k8s/src/main/scala/io/buoyant/k8s/IngressCache.scala

```diff
@@ -126,8 +132,12 @@ class IngressCache(namespace: Option[String], apiClient: Service[Request, Respon
         IngressPath(rule.host, path.path, namespace.getOrElse("default"), path.backend.serviceName, path.backend.servicePort)
       }
 
-      val fallback = spec.backend.map(b => IngressPath(None, None, namespace.getOrElse("default"), b.serviceName, b.servicePort))
-      IngressSpec(ingress.metadata.flatMap(_.name), namespace, fallback, paths)
+      val name = ingress.metadata.flatMap(_.name)
+      val fallback = spec.backend.map(b => IngressPath(None, None, namespace.getOrElse("default"), b.serviceName, b.servicePort)).filter { p =>
```

Comment:
I think this might be more clear written as:
```
if (strict) {
  spec.backend...
} else {
  None
}
```

Reply:
I'm afraid I don't quite follow the suggestion - I'm very new to Scala. Are you suggesting something like this?
```scala
val fallback =
  if (strict) spec.backend.map(b => ...)
  else None
```

That's how I had originally implemented this. I changed to the current implementation because I wanted to log in the `None` case and assumed (probably incorrectly) that I couldn't call `log.warning` from inside the assignment if/else.

Reply:
Scala is weird but you totally can!

```
val fallback = if (strict) {
  None
} else {
  log...
  spec.backend.map(b => ...)
}
```

Reply:
expression languages are wild

Reply:
I was in the process of changing this when I recalled another reason I ended up with the current implementation: determining the potential fallback path before making the decision whether to include it lets us log more specifically what default backend was omitted.
```
WARNING: ingress the-ingress.default: ignoreDefaultBackends mode enabled, ignoring default backend: IngressPath(None,None,default,fallback,2022)
```
Admittedly the log line would already mention the ingress resource name and namespace, so including the stringified `IngressPath` is arguably superfluous.

Does this change the verdict on the current implementation?

Reply:
it doesn't seem to me like additionally logging the IngressPath is very useful.

Reply:
Ack. Fixed!